packages/datastore/__tests__/DataStore.ts

```diff
@@ -2699,15 +2699,58 @@ describe('DataStore tests', () => {
 				);
 			});
 
-			test('valid model with nulls', () => {
-				expect(() => {
-					new Model({
-						field1: 'someField',
-						dateCreated: new Date().toISOString(),
-						emails: null,
-						ips: null,
-					});
-				}).not.toThrow();
+			test('valid model with null optional fields', () => {
+				const m = new Model({
+					field1: 'someField',
+					dateCreated: new Date().toISOString(),
+					optionalField1: null,
+				});
+				expect(m.optionalField1).toBe(null);
+			});
+
+			test('valid model with `undefined` optional fields', () => {
+				const m = new Model({
+					field1: 'someField',
+					dateCreated: new Date().toISOString(),
+					optionalField1: undefined,
+				});
+				expect(m.optionalField1).toBe(null);
+			});
+
+			test('valid model with omitted optional fields', () => {
+				const m = new Model({
+					field1: 'someField',
+					dateCreated: new Date().toISOString(),
+					/**
+					 * Omitting this:
+					 *
+					 * optionalField: undefined
+					 */
+				});
+				expect(m.optionalField1).toBe(null);
+			});
+
+			test('copyOf() setting optional field to null', () => {
+				const original = new Model({
+					field1: 'someField',
+					dateCreated: new Date().toISOString(),
+					optionalField1: 'defined value',
+				});
+				const copied = Model.copyOf(original, d => (d.optionalField1 = null));
+				expect(copied.optionalField1).toBe(null);
```

Comment:
I already tested this locally, so I know it works as expected, but I think we should also ensure that  optional fields with a value that are omitted from the `copyOf` aren't accidentally nullified.

```suggestion
				const emailsVal = ['test@test.test'];
				const original = new Model({
					field1: 'someField',
					dateCreated: new Date().toISOString(),
					optionalField1: 'defined value',
					emails: emailsVal,
				});
				const copied = Model.copyOf(original, d => (d.optionalField1 = null));
				expect(copied.optionalField1).toBe(null);
				expect(copied.emails).toEqual(emailsVal);
```